### PR TITLE
BFD-2021 add: x-ray policy to server roles

### PR DESCRIPTION
**Pull request by kcirtapfromspace**
**JIRA Ticket:**
[BFD-2021](https://jira.cms.gov/browse/BFD-2021)

**User Story or Bug Summary:**

As a BFD Engineer, I want a single dashboard that reliably reports the overall system's availability, defined as the system's ability to perform primary and ancillary functions as expected. To enable that holistic view please enable cloudwatch agent functionality to consume OpenTelemetry metrics. 

---

### What Does This PR Do?

Adds Permissions to server roles to enable output to x-ray service. 

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/terraform/modules/resources/iam/main.tf [hunk 1]
```diff
@@ -102,8 +102,37 @@ resource "aws_iam_policy" "ssm" {
...
+                "xray:GetSamplingRules",
+                "xray:GetSamplingTargets",
+                "xray:GetSamplingStatisticSummaries"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+}
```
Comment (mjburling) [inline ops/terraform/modules/resources/iam/main.tf:137]: Looking over the documentation and available IAM resources, these policies are no different from the AWS Managed `arn:aws:iam::aws:policy/AWSXRayDaemonWriteAccess` policy. However, we last discussed potentially tightening this policy to restrict beyond i.e. `"Resource": "*"`. If we're not restricting any further, I think we ought to favor the AWS Managed policy here and in the `migrator` module.

File: ops/terraform/modules/resources/iam/main.tf [hunk 2]
```diff
@@ -102,8 +102,37 @@ resource "aws_iam_policy" "ssm" {
 EOF
 }
 
-# attach AWS managed CloudWatchAgentServerPolicy to all EC2 instances
+# attach AWS managed SSM paramaters to all EC2 instances
```
Comment (mjburling) [inline ops/terraform/modules/resources/iam/main.tf:105]: 👍 Thanks for catching the copy/pasta error. Let's solve for the typo while we're at it.
```suggestion
# attach AWS managed SSM parameters to all EC2 instances
```

**Review by mjburling — CHANGES_REQUESTED** [review]

File: ops/terraform/modules/resources/iam/main.tf (hunk 1 above)
Comment (kcirtapfromspace) [reply ops/terraform/modules/resources/iam/main.tf:137]: Good call out, didn't realize there was a managed policy. None of those permissions allow for Resource types beyond *(required) or additional restrictions via condition tags.

File: ops/terraform/modules/resources/iam/main.tf (hunk 2 above)
Comment (kcirtapfromspace) [reply ops/terraform/modules/resources/iam/main.tf]: Will fix (mainly by making sure my spell check is turned on for all file type) Thanks for the keen eye. 😃

**Review by mjburling — CHANGES_REQUESTED** [review]
Just a couple gripes with extra whitespace, but otherwise this looks just fine!

Despite that adoption of AWS X-Ray's still being in somewhat of a POC phase for BFD, the _extended_ permissions here are safe and easily reversed. At worst, they will go unused in some cases while the solution is evaluated before making a more informed decisions on the near-term observability direction.

**Review by mjburling — APPROVED** [review]

**Review by mjburling — APPROVED** [review]

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**